packages/eslint-plugin/src/rules/no-explicit-any.ts

```diff
@@ -170,36 +176,88 @@ export default createRule<Options, MessageIds>({
       );
     }
 
+    /**
+     * Checks if the node is within a keyof any expression
+     * @param node the node to be validated.
+     * @returns true if the node is within a keyof any expression, false otherwise
+     * @private
+     */
+    function isNodeWithinKeyofAny(node: TSESTree.Node): boolean {
+      return (
+        node.parent?.type === AST_NODE_TYPES.TSTypeOperator &&
+        node.parent.operator === 'keyof'
+      );
+    }
+
+    /**
+     * Creates a fixer that replaces a keyof any with PropertyKey
+     * @param node the node to be fixed.
+     * @returns a function that will fix the node.
+     * @private
+     */
+    function createPropertyKeyFixer(node: TSESTree.Node) {
+      return (fixer: TSESLint.RuleFixer) => {
+        return fixer.replaceText(
+          node.parent as TSESTree.TSTypeOperator,
+          'PropertyKey',
+        );
+      };
+    }
+
+    /**
+     * Creates a fixer that replaces any with unknown
+     * @param node the node to be fixed.
+     * @returns a function that will fix the node.
+     * @private
+     */
+    function createUnknownFixer(node: TSESTree.Node) {
+      return (fixer: TSESLint.RuleFixer): TSESLint.RuleFix => {
+        return fixer.replaceText(node, 'unknown');
+      };
+    }
+
     return {
       TSAnyKeyword(node): void {
+        const isKeyofAny = isNodeWithinKeyofAny(node);
+
         if (ignoreRestArgs && isNodeDescendantOfRestElementInFunction(node)) {
           return;
         }
 
+        const propertyKeySuggestion: TSESLint.SuggestionReportDescriptor<MessageIds> =
+          {
+            messageId: 'suggestPropertyKey',
+            fix: createPropertyKeyFixer(node),
+          };
+
+        const unknownSuggestion: TSESLint.SuggestionReportDescriptor<MessageIds> =
+          {
+            messageId: 'suggestUnknown',
+            fix: createUnknownFixer(node),
+          };
+
+        const neverSuggestion: TSESLint.SuggestionReportDescriptor<MessageIds> =
+          {
+            messageId: 'suggestNever',
+            fix(fixer: TSESLint.RuleFixer): TSESLint.RuleFix {
+              return fixer.replaceText(node, 'never');
+            },
+          };
```

Comment:
[Refactor] Could you move these back inline, please? These objects don't need to be created until & unless they're needed. Plus this is a lot of explicit typing that can be inferred.

Reply:
This request is still open, did you mean to push some local code? 🙂